pkg/monitor/cluster/certificateexpirationstatuses.go

```diff
@@ -0,0 +1,95 @@
+package cluster
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/genevalogging"
+	"github.com/Azure/ARO-RP/pkg/util/dns"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+const (
+	certificateExpirationMetricName = "certificate.expirationdate"
+	secretMissingMetricName         = "certificate.secretnotfound"
+)
+
+func (mon *Monitor) emitCertificateExpirationStatuses(ctx context.Context) error {
+	// report NotAfter dates for Ingress and API (on managed domains), and Geneva (always)
+	var certs []*x509.Certificate
+
+	mdsdCert, err := mon.getCertificate(ctx, operator.Namespace, operator.SecretName, genevalogging.GenevaCertName)
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return err
+		}
+		mon.emitGauge(secretMissingMetricName, int64(1), map[string]string{
+			"secretMissing": operator.SecretName,
+		})
+	} else {
+		certs = append(certs, mdsdCert)
+	}
```

Comment:
Ok i'm going back on my original statement.  

We should align this with whatever we do https://github.com/Azure/ARO-RP/pull/2976/files#diff-316e632485d745cf8ccd3fb66e5c6dd71b3f33156040db64751d78e9596ad81aR58-R66

If you want to keep original structure, I'm fine with it.  I thought we could return early, but seeing as we can't now, let's just align the two.

Reply:
insert :old-man-yells-at-ben: emoji :laughing:

Reply:
I think the chained `else if` is how I've done it other places. Maybe a switch statement or something would look better for multiple ways of handling 🤷.